### ISSUE TITLE
🐛 `noInvalidGridAreas` report value from other properties.

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#3827]
### Environment information

```block
Main
```

### What happened?

The rule reports provide diagnostics for many properties that are not related to grid.

On example in the [playground](https://biomejs.dev/playground/?lintRules=all&files.main.css=IwBjAGEAcgBiAG8AbgBhAGQAcwAgAHsACgAgACAAZgBvAG4AdAAtAGYAYQBtAGkAbAB5ADoAIAAtAGEAcABwAGwAZQAtAHMAeQBzAHQAZQBtACwAIABCAGwAaQBuAGsATQBhAGMAUwB5AHMAdABlAG0ARgBvAG4AdAAsACAAIgBTAGUAZwBvAGUAIABVAEkAIgAsACAAUgBvAGIAbwB0AG8ALAAgAE8AeAB5AGcAZQBuAC0AUwBhAG4AcwAsACAAVQBiAHUAbgB0AHUALAAgAEMAYQBuAHQAYQByAGUAbABsACwAIAAiAEgAZQBsAHYAZQB0AGkAYwBhACAATgBlAHUAZQAiACwAIABIAGUAbAB2AGUAdABpAGMAYQAsACAAQQByAGkAYQBsACwAIABzAGEAbgBzAC0AcwBlAHIAaQBmADsACgB9AA%3D%3D)

I found this issue by analyzing regressions in our Ecosystem CI. For example the [test for the `saadeghi/daisyui` repository](https://github.com/biomejs/ecosystem-ci/actions/runs/10753292658/job/29822625772) emitted 22 diagnostics for this rule.

### Expected result

The rule should not report properties unrelated to grids.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
